src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -7760,10 +9220,18 @@ public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y)
             public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y) => x * y;
 #endif
 
-            public static float Invoke(Vector128<float> x) => HorizontalAggregate<MultiplyOperator>(x);
-            public static float Invoke(Vector256<float> x) => HorizontalAggregate<MultiplyOperator>(x);
```

Comment:
We can't just update HorizontalAggregate to do this in one place rather than adding all these shuffles? e.g.
```C#
private static float HorizontalAggregate<TAggregate>(Vector128<float> x) where TAggregate : struct, IBinaryOperator
{
    // We need to do log2(count) operations to compute the total sum
    x = TAggregate.Invoke(x, Vector128.Shuffle(x, Vector128.Create(2, 3, 0, 1)));
    x = TAggregate.Invoke(x, Vector128.Shuffle(x, Vector128.Create(1, 0, 3, 2));
    return x.ToScalar();
}
```

Reply:
Yeah, we could probably share it like this. Will update

Reply:
Fixed.